File: csharp/FizzBuzz/Tests/FizzBuzzTest.cs
using FluentAssertions;
using Implementation;
using Xunit;

namespace Tests
{
    public class FizzBuzzTest
    {
        private readonly FizzBuzzEngine _target;

        public FizzBuzzTest()
        {
            _target = new FizzBuzzEngine();
        }

        [Theory]
        [InlineData(0, "0")]
        [InlineData(1, "1")]
        public void CalculcateShouldReturnNormalNumbersAsIs(int number, string expectedResult)
        {
            // act
            var result = _target.Calculate(number);

            // assert
            result.Should().Be(expectedResult);
        }

        [Theory]
        [InlineData(3)]
        [InlineData(6)]
        public void CalculateShouldReturnFizz(int number)
        {
            // act
            var result = _target.Calculate(number);

            // assert
            result.Should().Be("Fizz");
        }
    }
}

File: php/FizzBuzz/tests/FizzBuzzTest.php
<?php
namespace Core\tests;

use Core\FizzFuzzEngine;
use PHPUnit_Framework_TestCase;

class FizzBuzzTest extends PHPUnit_Framework_TestCase
{
    /**
     * @var FizzFuzzEngine
     */
    private $target;

    protected function setUp()
    {
        $this->target = new FizzFuzzEngine();
    }

    public function fizzBuzzProvider()
    {
        return [
            [1, "1"],
            [2, "2"],
            [3, "Fizz"],
            [6, "Fizz"],
            [5, "Buzz"],
            [10, "Buzz"],
            [15, "Fizz-Buzz"],
        ];
    }

    /**
     * @dataProvider fizzBuzzProvider
     */
    public function testThatFizzBuzzEngineReturnsExpectedResults($number, $expected)
    {
        // act
        $result = $this->target->calculateResult($number);
        // assert
        $this->assertThat($result, self::equalTo($expected));
    }
}
